lib/timeline/timeline.go

```diff
@@ -0,0 +1,280 @@
+/*
+Copyright 2019 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package timeline provides interfaces for keeping track of cluster status events.
+package timeline
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	pb "github.com/gravitational/satellite/agent/proto/agentpb"
+)
+
+// Stamp defines default timestamp format for events.
+const Stamp = "Jan _2 15:04:05"
+
+// Event represents a Timeline event. An event occurrs  whenever there is a
+// change in the cluster status. An event could be triggered by a failed
+// heartbeat or a failed health check.
+type Event struct {
+	// TimeStamp specifies when the event occurred.
+	TimeStamp time.Time
+
+	// TODO: can we rely on user CLI sessions to provide colored text?
+	// If not, maybe we should indicate the different event types a different
+	// way.
+
+	// Color specifies the color of the event.
+	// Red -> Indicates change has caused state to degraded.
+	// Yellow -> Indicates change has not caused any state changes.
+	// Green -> Indicates changes has caused state to recover.
+	Color string
```

Comment:
Can we reuse existing probe statuses instead of introducing another thing to basically represent the same thing? Then the caller can output them appropriately (provided we return structured response, see comment above).